### PR TITLE
[WIP] refactor: Reformat the openlibrary/ directory with black

### PR DESCRIPTION
**Pull request by dhruvmanila**
Subtask of #4776 

Run `black` on ***only*** `openlibrary/` directory:
```console
> black --skip-string-normalization openlibrary/
```

**TODO:**
- [x] Review the initial pass by `black`
- [ ] List out any formatting which does not seem proper
- [ ] Ignore this commit in `.git-blame-ignore-revs` similar to https://github.com/internetarchive/infogami/pull/169

Closes #

### Stakeholders

@cclauss @cdrini

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,7 +70,7 @@ lint-diff:
 
 lint:
 	# stop the build if there are Python syntax errors or undefined names
-	$(PYTHON) -m flake8 . --count --exclude=$(FLAKE_EXCLUDE) --select=E9,F63,F7,F82 --show-source --statistics
+	$(PYTHON) -m flake8 . --count --exclude=$(FLAKE_EXCLUDE) --ignore=E9,F63,F7,F82 --show-source --statistics
```
Comment (cclauss) [inline Makefile:73]: ??? We should ignore only those tests that we still fail.  These are tests that we should definitely should _not_ fail.

Comment (dhruvmanila) [reply Makefile:73]: Oops, sorry. Completely misread the title.

Comment (dhruvmanila) [reply Makefile:73]: I will revert the last commit but the failure it in the `lint-diff` step:
```
58    E203 whitespace before ':'
2     E265 block comment should start with '# '
1     E402 module level import not at top of file
150   E501 line too long (98 > 88 characters)
3     E741 ambiguous variable name 'l'
2     F405 'translate' may be undefined, or defined from star imports: openlibrary.catalog.marc.fast_parse
4     F601 dictionary key 'resource_type' repeated with different values
5     F841 local variable 'cat' is assigned to but never used
273   W503 line break before binary operator
17    W605 invalid escape sequence '\d'
```
W503 should be ignored: https://black.readthedocs.io/en/stable/the_black_code_style.html#line-breaks-binary-operators

E203 should be ignored as well: https://black.readthedocs.io/en/stable/the_black_code_style.html#slices

Should we increase the line length to 100?

The rest should probably be fixed manually.

Comment (cclauss) [reply Makefile:73]: Increase `--max-line-length=` until the E501 errors go away and then we can create a __future__ issue to wrap lines to 88.

File: openlibrary/catalog/amazon/crawl.py [hunk 2]
```diff
@@ -173,18 +209,19 @@ def get_cats(root):
...
     return cats
 
     for e in container.find('table').find_class('refinementLink'):
         a = e.getparent()
         assert a.tag == 'a'
-        cat = { 'url': a.attrib['href'], 'title': e.text }
+        cat = {'url': a.attrib['href'], 'title': e.text}
         href = a.attrib['href']
         m = re_rh_n.search(href)
         cats.append((int(m.group(1)), e.text))
```
Comment (dhruvmanila) [inline openlibrary/catalog/amazon/crawl.py:222]: This code is unreachable due to the return statement above. Removing it ...

**Comment by cclauss** [comment]
`--max-line-length=1195`
> openlibrary/utils/olcompress.py:10:89: E501 line too long (1195 > 88 characters)

**Comment by dhruvmanila** [comment]
Hey, @cdrini Can you please take a look at why the tests are failing? https://github.com/internetarchive/openlibrary/pull/5079/checks?check_run_id=2498124543#step:8:124

This function was added by you so you would know more about it :)

**Pull request closed without merging**